smart_open/bytebuffer.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+"""Implements a byte buffer that allows callers to read data with minimal
+copying, and has a fast __len__ method. The buffer is parametrized by its
+chunk_size, which is the number of bytes that it will read in from the supplied
+reader or iterable when the buffer is being filled. As primary use case for this
+buffer is to amortize the overhead costs of transferring data over the network
+(rather than capping memory consumption), it leads to more predictable
+performance to always read the same amount of bytes each time the buffer is
+filled, hence the chunk_size parameter instead of some fixed capacity.
+
+The bytes are stored in a bytestring, and previously-read bytes are freed when
+the buffer is next filled (by slicing the bytestring into a smaller copy)."""
+
+class ByteBuffer(object):
+    def __init__(self, chunk_size):
```

Comment:
A sensible default value is helpful here:

```suggestion
    def __init__(self, chunk_size=io.DEFAULT_BUFFER_SIZE):
```

Reply:
Fixed in https://github.com/RaRe-Technologies/smart_open/pull/284/commits/2e88391c577a0cbd3c0e22b96765cae653707dab